x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -57,7 +60,7 @@ synchronized void shutDown() {
         this.state.set(WatcherState.STOPPING);
         shutDown = true;
         clearAllocationIds();
-        watcherService.shutDown();
+        watcherService.shutDown(() -> {});
```

Comment:
@martijnvg pointed out that we may want to push the STOPPED state to inside the callback. That sounds right, but will need to ensure that delaying the STOPPED state won't cause issues.

Reply:
this has been implemented in https://github.com/elastic/elasticsearch/pull/43939/commits/c54c7da61bd6ce96ac4dda3f79084a0e8886815e